### PR TITLE
Caret position tracking within iframe editor

### PR DESCRIPTION
**Pull request by gallyamb**
- Separate handlers for keypress and keyup/keydown events.
 - Add caretPosition tracking within iframe editor.
 - Refactoring.

### CODE REVIEW

File: player/src/main/java/com/focusit/jsflight/player/webdriver/SeleniumDriver.java [hunk 1]
```diff
@@ -226,68 +286,59 @@ public WebDriver getDriverForEvent(JSONObject event, BrowserType browserType, St
...
-                    if (proxyPort != 0)
-                    {
-                        host += ":" + proxyPort;
-                    }
-                    Proxy proxy = new Proxy();
-                    proxy.setHttpProxy(host).setFtpProxy(host).setSslProxy(host);
-                    cap.setCapability(CapabilityType.PROXY, proxy);
+                    host += ":" + proxyPort;
                 }
+                Proxy proxy = new Proxy();
```
Comment (d0k1) [inline player/src/main/java/com/focusit/jsflight/player/webdriver/SeleniumDriver.java]: is a proxy with no port specified usable?
As for me it is not

File: player/src/main/java/com/focusit/jsflight/player/webdriver/SeleniumDriver.java [hunk 2]
```diff
@@ -369,98 +420,112 @@ public WebElement apply(WebDriver input)
...
         if (skipKeyboardForElement(element))
         {
             LOG.warn("Keyboard processing for non empty Date is disabled");
             return;
         }
 
-        if (EventType.KEY_PRESS.equalsIgnoreCase(event.getString(EventConstants.TYPE)))
+        if (!event.has(EventConstants.CHAR))
+        {
+            throw new IllegalStateException("Keypress event don't have a char");
```
Comment (d0k1) [inline player/src/main/java/com/focusit/jsflight/player/webdriver/SeleniumDriver.java]: will it break an ongoing playback?

File: player/src/main/java/com/focusit/jsflight/player/webdriver/SeleniumDriver.java (hunk 1 above)
Comment (gallyamb) [reply player/src/main/java/com/focusit/jsflight/player/webdriver/SeleniumDriver.java]: Well, I haven't changed anything here except delete try / catch. I agree with you. I'll fix it

File: player/src/main/java/com/focusit/jsflight/player/webdriver/SeleniumDriver.java (hunk 2 above)
Comment (gallyamb) [reply player/src/main/java/com/focusit/jsflight/player/webdriver/SeleniumDriver.java]: No, playback will continue. But we'll see the error in the logs

Comment (d0k1) [reply player/src/main/java/com/focusit/jsflight/player/webdriver/SeleniumDriver.java]: it's okay

**Comment by gallyamb** [comment]
@d0k1 please, review the changes. I've added `Organize dependencies` commit here just for convinience.

**Comment by d0k1** [comment]
it seems OK, you're welcome to merge this pr

**Pull request merged**